### PR TITLE
Update "Bootstrapping" chapter

### PR DESCRIPTION
**Pull request by IndyIndyIndy**

### CODE REVIEW

File: Documentation/ApiOverview/Bootstrapping/Index.rst [hunk 1]
```diff
@@ -7,116 +7,112 @@ Bootstrapping
...
+There are four types of Applications provided by the TYPO3 Core:
 
-   This bootstrapping API is internal and may change at any time in the near future
-   even in minor updates. It is thus discouraged to use it in third party code.
-   Use this class only if other extensibility possibilities such as
-   :ref:`Hooks <hooks>`, Signals or :ref:`XCLASS <xclasses>`
-   are not enough to reach your goals.
+\\TYPO3\\CMS\\Frontend\\Http\\Application
+-----------------------------------------
+All incoming web requests coming to index.php in the main directory, handling all TSFE and eID requests.
```
Comment (sypets) [inline Documentation/ApiOverview/Bootstrapping/Index.rst:26]: I would use a full sentence here. How about

> This class handles all incoming web requests ...

File: Documentation/ApiOverview/Bootstrapping/Index.rst [hunk 2]
```diff
@@ -7,116 +7,112 @@ Bootstrapping
...
+The Application sets TYPO3_MODE=FE very early.
+The Application checks if all configuration is given, otherwise redirects to the TYPO3 Install Tool.
 
+\\TYPO3\\CMS\\Backend\\Http\\Application
+----------------------------------------
+All incoming web requests for any regular Backend call inside typo3/\*. This handles three types of Request Handlers:
```
Comment (sypets) [inline Documentation/ApiOverview/Bootstrapping/Index.rst:32]: e.g. 

> This class handles all incoming web request ... Three types of request handlers are registered:

File: Documentation/ApiOverview/Bootstrapping/Index.rst [hunk 3]
```diff
@@ -7,116 +7,112 @@ Bootstrapping
...
+- The Backend Module Request Handler, which handles all types of modules triggered on requests with an "M" GET Parameter
+- The regular Request handler for typical other backend calls on index.php.
 
-   use TYPO3\CMS\Core\Core\Bootstrap;
+The Application checks if all configuration is given, otherwise redirects to the TYPO3 Install Tool.
 
-   ###
+\\TYPO3\\CMS\\Backend\\Console\\Application
+-------------------------------------------
+All CLI Requests handled by cli_dispatch.php. Only executes the parts that are necessary for Backend CLI Scripts used
```
Comment (sypets) [inline Documentation/ApiOverview/Bootstrapping/Index.rst:42]: suggestion:

> The backend console Application handles all cli request initiated by ...

**Comment by lolli42** [comment]
Who would like to update this one?

File: Documentation/ApiOverview/Bootstrapping/Index.rst (hunk 2 above)
Comment (linawolf) [reply Documentation/ApiOverview/Bootstrapping/Index.rst:32]: there is actually just one in the current master

**Comment by linawolf** [comment]
commits moved to #1225 so they are easier to edit for us

**Pull request closed without merging**